### PR TITLE
Changed passing children by props to nest between tags in docs

### PR DESCRIPTION
**Pull request by goofynugtz**
### Initial checklist

*   [x] I read the support docs 
*   [x] I read the contributing guide 
*   [x] I agree to follow the code of conduct 
*   [x] I searched issues and couldn’t find anything (or linked relevant results below) 
*   [x] If applicable, I’ve added docs and tests

### Description of changes
Some of the examples mentioned in the readme are passing children as props which gives a ESLint error of `react/no-children-prop` on Next.js 13 with ESLint configured
```javascript
<ReactMarkdown children={markdown}  />
```
```
yarn run v1.22.19
$ next build

Failed to compile.

./pages/blogs/[slug].tsx
72:13  Error: Do not pass children as props. Instead, nest children between the opening and closing tags.  react/no-children-prop

info  - Need to disable some ESLint rules? Learn more here: https://nextjs.org/docs/basic-features/eslint#disabling-rules
error Command failed with exit code 1.
```
Updated the examples to nest child component between the opening and closing tags.
```javascript
<ReactMarkdown>{markdown}</ReactMarkdown>
```

### CODE REVIEW

File: readme.md
Comment (remcohaszing) [inline readme.md:146]:
```suggestion
  <ReactMarkdown remarkPlugins={[remarkGfm]}>
```

File: readme.md
Comment (remcohaszing) [inline readme.md:254]:
```suggestion
  <ReactMarkdown remarkPlugins={[remarkGfm]}>{markdown}</ReactMarkdown>,
```

File: readme.md
Comment (remcohaszing) [inline readme.md:411]:
```suggestion
  >
    {`The lift coefficient ($C_L$) is a dimensionless coefficient.`}
  </ReactMarkdown>,
```

File: readme.md
Comment (remcohaszing) [inline readme.md:372]:
```suggestion
    {markdown}
```

File: readme.md
Comment (remcohaszing) [inline readme.md:529]:
```suggestion
  <ReactMarkdown rehypePlugins={[rehypeRaw]}>{input}</ReactMarkdown>,
```

**Review by remcohaszing — CHANGES_REQUESTED** [review]
Thanks! I agree passing children as actual children instead of a prop is better. I just have some nits about spacing.

File: readme.md
Comment (JounQin) [reply readme.md:411]: This can be plain text without template string?

```jsx
  <ReactMarkdown
    remarkPlugins={[remarkMath]}
    rehypePlugins={[rehypeKatex]}
  >
    The lift coefficient ($C_L$) is a dimensionless coefficient.
  </ReactMarkdown>
```

Comment (remcohaszing) [reply readme.md:411]: It could be, but the spacing rules for JSX often lead to confusion. By this I don’t mean to say we shouldn’t use that, just that there is an argument to make for not doing it.

**Comment by goofynugtz** [comment]
Thanks for the reviews. I have made the suggested changes.

**Review by remcohaszing — APPROVED** [review]

**Review by ChristianMurphy — COMMENTED** [review]
I'm cautious about this.
Yes as a general react practice, using nesting over the `children` prop is ideal.
That said, JSX spacing rules and template string spacing rules inside JSX are a frequent source of confusion for new adopters https://github.com/remarkjs/react-markdown/issues?q=is%3Aissue+JSX+is%3Aclosed

I'd lean against making this change.
I think it will cause more confusion than it clarifies.
But will not block the PR if other maintainers feel strongly otherwise.

**Comment by remcohaszing** [comment]
If we don’t want this, I think we should rename the prop to something else, e.g. `content` or `markdown`. (A semver major is upcoming anyway.)

**Comment by ChristianMurphy** [comment]
> If we don’t want this, I think we should rename the prop to something else, e.g. `content` or `markdown`. (A semver major is upcoming anyway.)


I don't think it necessitates a major rework of the API.
There is a lot of overlap in adopters who don't understand that the two forms of children in react are equivalent and those don't understand the rules of JSX, and vice versa.

We could leave the docs as-is and let adopters choose which syntax they prefer.

---

To per-emptively answer the question: "could we document the rules of JSX next to the example?"
Yes we could, but it wouldn't help much, no matter how many ":warning: IMPORTANT READ THIS!" we put around the example, many newcomers will ignore it, copy the example, and open an issue asking why it doesn't work as they'd expect (having not read the JSX spacing rules).

**Comment by wooorm** [comment]
Did y’all see https://github.com/remarkjs/react-markdown/issues/749#issuecomment-1636768884?

I think people are going to try to pass children whether it works or not. As “text” inside the component, or as a sole expression inside the component. The main similar project to this, `markdown-to-jsx`, uses `children` too. And so do several other similar projects. Previously we used a different field, but a couple majors back, we moved to `children`, because it’s a common expectation from users.

Theoretically, we could throw on that. Or we could ignore it and support another field. I don’t prefer those over the current, sole, `children`.

The differences between `<ReactMarkdown children={x} />`, `<ReactMarkdown>{x}</ReactMarkdown>`, and `<ReactMarkdown># literal X</ReactMarkdown>`, is a bit complex for (somewhat) newcomers to React. But I feel that’s more a React fact. That it’s their responsibility. Which is fine to document with an example appendix *and* improved examples as in GH-749.

I also mostly agree with Christian! (only point that I think we can use *one* style in the examples, so people copy the nicest one, where currently 2 are shown)

**Comment by wooorm** [comment]
Used one style in the docs: `<Markdown>{markdown}</Markdown>`, so that solves this!

**Pull request closed without merging**
